doc/webservices/extract_api.rst

```diff
@@ -0,0 +1,478 @@
+
+This service is a paid service. Each invoice processing will cost you one credit. Three packs of different size can be bought on `iap.odoo.com <https://iap.odoo.com/iap/in-app-services/259?sortby=date>`_.
+
+You can either use this service directly in the Odoo Accounting App or by using the API. The Extract API which is detailed in the next section allows you to integrate our service directly in your own projects.
+
+Invoices
+========
+The extract API can be contacted at the following JSON-RPC2_ endpoint: `https://iap-extract.odoo.com`
+
+Expected successful flow
+------------------------
+
+1. Call :ref:`/iap/invoice_extract/parse <webservices/extract_api/invoice_parse>` to submit your invoices (one call for each invoice). On success you receive a `document_id` in the response. 
+2. You then have to regularly poll the :ref:`/iap/invoice_extract/get_results <webservices/extract_api/invoice_get_results>` route to get the document's parsing status.
+3. Once the result received you can validate it by calling :ref:`/iap/invoice_extract/validate <webservices/extract_api/invoice_validate>` and sending the expected values. This step is optional but greatly helps the system to improve.
+
+These 3 routes are detailed in the next section. A python implementation of the full flow and a token for integration testing are provided at the end of the page.
+
+Routes
+------
+.. _webservices/extract_api/invoice_parse:
+
+/iap/invoice_extract/parse (POST)
+'''''''''''''''''''''''''''''''''
+
+Description
+^^^^^^^^^^^
+Request a processing of the document from the OCR. The route will return a `document_id` you can use to obtain the result of your request.
+
+Request Body
+^^^^^^^^^^^^
+
+* ``jsonrpc``: A string specifying the version of the JSON-RPC protocol. Must be exactly “2.0”. **REQUIRED**
+* ``method``: A string containing the name of the method to be invoked. Must be “call”. **REQUIRED**
+* ``params``:
+    * ``account_token``: token of the account from which credits will be taken. Each successful call cost one token. **REQUIRED**
+    * ``version``: The version will determine the format of your requests and the format of the server response. Some results can be unavailable in older versions. For the current version 1.2.0, send ‘120’. If not specified, the latest version will be used. **OPTIONAL**.
+    * ``documents``: The invoice must be provided as a string in the ASCII format. The list should contain only one string. If multiple strings are provided only the first string corresponding to a pdf will be processed. If no pdf is found, the first string will be processed. This field is a list only for legacy reasons. The supported extensions are *pdf*, *png*, *jpg* and *bmp*. **REQUIRED**
+    * ``user_infos``: 
+        Informations concerning the person to whom the invoice is intended. These informations are not required in order for the service to work but they greatly improve the performances. **REQUIRED** (can be empty).
```

Comment:
**Information** concerning the person to whom the invoice is intended. **This information is** not required in order for the service to work but **it greatly improves** the **performance**.